micrometer-core/src/main/java/io/micrometer/core/instrument/binder/kafka/KafkaMetrics.java

```diff
@@ -87,7 +91,7 @@ else if (START_TIME_METRIC_NAME.equals(name.name()))
         }
         if (startTime != null) bindMeter(registry, startTime, meterName(startTime), meterTags(startTime));
         // Collect dynamic metrics
-        checkAndBindMetrics(registry);
+        scheduler.scheduleAtFixedRate(() -> checkAndBindMetrics(registry), 0, 5, SECONDS);
```

Comment:
This looks good to start with!

I was wondering two additional things to consider here:
- If number of metrics eventually stabilize, we could consider to have a back-off here to query less frequently if we get all metrics registered. Although this might not fit well with applications that have a more dynamic behavior (e.g. subscribing/sending to topics by pattern)
- We could customize how frequent this rate should be by adding a input parameter maybe? This way if the application has multiple topics, then we can increase this.

Reply:
A backoff strategy is an interesting idea. I'm not sure it is worth the complexity until we have more reason for it. If the registry is a step registry, I would use the step as the interval. Otherwise, we can default to 1 minute (our general default step).

Reply:
Made a configuration parameter + change default value